pkg/operation/shoot/shoot.go

```diff
@@ -149,7 +149,11 @@ func (b *Builder) WithDefaultDomains(defaultDomains []*gardenerutils.Domain) *Bu
 
 // Build initializes a new Shoot object.
 func (b *Builder) Build(ctx context.Context, c client.Reader) (*Shoot, error) {
-	shoot := &Shoot{}
+	var (
+		shoot    = &Shoot{}
+		secret   *corev1.Secret
+		networks *Networks
```

Comment:
Similar you do not need a variable for networks if you move the initialization of shoot.Networks within the if block